source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -421,13 +421,22 @@ namespace
             PvtPrim* assignPrim = stage->createPrim(parent->getPath(), RtToken(matAssign->getName()), RtMaterialAssign::typeName());
             RtMaterialAssign rtMatAssign(assignPrim->hnd());
             
-            PvtPrim* collection = findPrimOrThrow(RtToken(matAssign->getCollectionString()), parent);
-            rtMatAssign.getCollection().addTarget(collection->hnd());
+            if (!matAssign->getCollectionString().empty()) {
+                PvtPrim* collection = findPrimOrThrow(RtToken(matAssign->getCollectionString()), parent);
+                rtMatAssign.getCollection().addTarget(collection->hnd());
+            }
+
+            if (!matAssign->getMaterial().empty()) {
+                PvtPrim* material = findPrimOrThrow(RtToken(matAssign->getMaterial()), parent);
+                rtMatAssign.getMaterial().addTarget(material->hnd());
+            }
 
-            PvtPrim* material = findPrimOrThrow(RtToken(matAssign->getMaterial()), parent);
-            rtMatAssign.getMaterial().addTarget(material->hnd());
+            if (matAssign->hasAttribute(MaterialAssign::EXCLUSIVE_ATTRIBUTE)) {
```

Comment:
Attribute is optional and will throw if string is empty. Check beforehand for presence of the attribute.